apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/LocationHelperModule.kt

```diff
@@ -0,0 +1,219 @@
+package com.ledger.live;
+
+import android.Manifest
+import android.app.Activity
+import android.content.Intent
+import android.util.Log
+
+// Native module imports
+import com.facebook.react.bridge.*
+import com.facebook.react.modules.core.DeviceEventManagerModule
+
+// Location imports
+import com.google.android.gms.common.api.ResolvableApiException
+import com.google.android.gms.location.*
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+
+import com.ledger.live.Constants.REQUEST_ENABLE_LOCATION
+
+/**
+ * Android Native module that provides:
+ * - a method to check and enable if necessary the location service
+ * - a method to listen to the state (enabled/disabled) of the location service
+ *
+ * See from https://developers.google.com/android/reference/com/google/android/gms/location/package-summary :
+ * - SettingsClient which provides a way to interact with the location settings-enabler APIs
+ * - FusedLocationProviderClient which provides a way to listen to the location service state updates
+ */
+class LocationHelperModule(private val reactContext: ReactApplicationContext) :
+    ReactContextBaseJavaModule(reactContext) {
+
+    // Resolves with a success or error value. reject is not used because we want to pass an
+    // information to the JS code.
+    private var locationPromise: Promise? = null
+
+    // LocationRequest used by both LocationSettings and FusedLocationProviderClient
+    private var locationRequest: LocationRequest = LocationRequest.create();
+
+    private var fusedLocationProviderClient: FusedLocationProviderClient;
+
+    // Listens to result from the location enable request activity.
+    // Request sent from requestEnableLocation.
+    private val activityEventListener = object : BaseActivityEventListener() {
+        override fun onActivityResult(
+            activity: Activity?,
+            requestCode: Int,
+            resultCode: Int,
+            data: Intent?
+        ) {
+            if (requestCode === REQUEST_ENABLE_LOCATION) {
+                when (resultCode) {
+                    Activity.RESULT_OK -> {
+                        locationPromise?.resolve(SUCCESS_LOCATION_ENABLED);
```

Comment:
here and below set back locationPromise to null no?

Reply:
Yes you're right !